src/megaapi_impl.cpp

```diff
@@ -2739,6 +2739,12 @@ void MegaTransferPrivate::setListener(MegaTransferListener *listener)
     this->listener = listener;
 }
 
+void MegaTransferPrivate::startRecursiveOperations(MegaRecursiveTransferOpertaion* op)
```

Comment:
As discussed IRL, this function should take `unique_ptr` to make ownership clear

Reply:
let's discuss with the team and see if we can find a way to do it that doesn't add too much syntax :-)

Reply:
I also prefer a clean syntax here. Overloading the code with (syntax) decorations is not always worthy if you are careful enough and add a comment about ownership close to the declaration of the function. Nevertheless, I understand the advantage of the `unique_ptr` pointed by Christian as well.

Reply:
If we add our own version of the make_unique function (which was added to the c++14 standard, but is just a (Ivaridically) templated function that would work in c++11 - and let's just add the most common case), that would allow us to cover cases like this with compiler-checked/managed ownership (and zero runtime overhead) without adding too much syntax - I'll try it and see how it looks - I'll post it in the cpp channel if it seems ok.

Reply:
`unique_ptr` is not just syntactic sugar: The compiler will be your friend and help you get ownership right. When the function takes a `unique_ptr` it is clear that the function will own the memory; no comments needed.